backend/src/services/aws/storage/storage.service.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  CreateBucketCommand,
+  CreateBucketCommandOutput,
+  DeleteBucketCommand,
+  DeleteBucketCommandOutput,
+  DeleteObjectCommand,
+  DeleteObjectCommandOutput,
+  GetObjectCommand,
+  GetObjectCommandOutput,
+  PutObjectCommand,
+  PutObjectCommandOutput,
+  S3Client,
+  S3ServiceException,
+} from '@aws-sdk/client-s3';
+
+import {
+  StorageManipulateFileRequestDto,
+  StorageUploadFileRequestDto,
+} from '~/common/types/types';
+import { StorageError } from '~/exceptions/exceptions';
+
+type Constructor = {
+  region: string;
+  credentials: {
+    accessKeyId: string;
+    secretAccessKey: string;
+  };
+};
+
+class Storage {
+  #storage: S3Client;
+
+  public constructor({ credentials, region }: Constructor) {
+    this.#storage = new S3Client({
+      region,
+      credentials,
+    });
+  }
+
+  public async createBucket(name: string): Promise<CreateBucketCommandOutput> {
+    try {
+      return await this.#storage.send(
+        new CreateBucketCommand({ Bucket: name }),
+      );
+    } catch (err) {
+      this.throwError(err);
+    }
+  }
+
+  public async deleteBucket(name: string): Promise<DeleteBucketCommandOutput> {
+    try {
+      return this.#storage.send(new DeleteBucketCommand({ Bucket: name }));
+    } catch (err) {
+      this.throwError(err);
+    }
+  }
```

Comment:
@what1s1ove not sure if we need create/delete bucket functionality 😕

Reply:
agreed. for now we need only upload an object